pkgs/applications/networking/cluster/levant/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "levant";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "hashicorp";
+    repo = "levant";
+    rev = "v${version}";
+    sha256 = "9M7a4i+DPKb1H9jOEVAvhvYxGwtj3dK/40n4GSy4Rqo=";
+  };
+
+  vendorSha256 = "5JlrgmIfhX0rPR72sUkFcofw/iIbIaca359GN9C9dhU=";
+
+  runVend = true;
+
+  # The tests try to connect to a Nomad cluster.
+  doCheck = false;
+
+  meta = with lib; {
+    description = "An open source templating and deployment tool for HashiCorp Nomad jobs";
+    homepage = "https://github.com/hashicorp/levant";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ max-niederman ];
+    platforms = platforms.unix;
```

Comment:
```suggestion
```
default of buildGoModule

Reply:
It's actually not. `buildGoModule` [inherits from](https://github.com/NixOS/nixpkgs/blob/2d47affbfb2d961b721a435513710ee7921d3dfd/pkgs/development/go-modules/generic/default.nix#L267) [Go](https://github.com/NixOS/nixpkgs/blob/2d47affbfb2d961b721a435513710ee7921d3dfd/pkgs/development/compilers/go/1.17.nix#L275), which is `platforms.linux ++ platforms.darwin`. I was able to verify this by checking the platforms with `nix-env -qa` with and without the platforms line. Admittedly, I have no idea whether the package would build on Cygwin, other BSDs, or Redox, but that's the platform [Nomad's package uses](https://github.com/NixOS/nixpkgs/blob/2d47affbfb2d961b721a435513710ee7921d3dfd/pkgs/applications/networking/cluster/nomad/generic.nix#L45).

Reply:
I think that's close enough. If it causes issues, we can revisit this.

Reply:
buildGoModule should not set platforms.unix or Linux + Darwin. The same is true for equivalent python and rust function. If those package sets ever support another platform we don't want to update hundreds of packages which just work on the new platform.

Reply:
I'm not totally sure what you mean. Are you saying that _consumers_ of `buildGoModule` shouldn't set `meta.platforms` or that `buildGoModule` itself shouldn't?